### PR TITLE
feat: download file with data streaming

### LINKED ISSUE #11: feat: download file
## Tasks

- [ ] Create the struct in [RMI](https://github.com/hawks-atlanta/rmi-java) repository
- [ ] Update the gateway [dev](https://github.com/hawks-atlanta/gateway-java/tree/dev) branch
- [ ] Implement the service [worker](https://github.com/hawks-atlanta/worker-java)

## Ideas

```
struct download {
     string uuid;
}
```

### PR DESCRIPTION
**Pull request by Woynert**
> [!IMPORTANT]  
> Required for https://github.com/hawks-atlanta/gateway-java/pull/46

## :sparkles: Includes

- Look up in files in backups.
- Between **Worker** and **Gateway** files are transferred through byte stream using RMIIO library. This allows to pass big files without loading them all at once in memory.
	
	> [!NOTE]  
	> Transfers between Gateway and other clients still require to load the
 whole file.
- Various tests `> 80%`.

## :canned_food: Test

Before running tests, remember to setup the directory structure as described in https://github.com/hawks-atlanta/worker-java/pull/14.

Then, run:

```sh
AVAILABLE_VOLUMES=1,2,3 VOLUME_BASE_PATH=</full/path/to/vols> gradle cleanTest test
```

### CODE REVIEW

File: app/src/main/java/worker/file/DownloadFile.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		RemoteInputStreamServer istream = null;
+
+		try {
+
+			File file = new File (String.format (
+				"%s/files/volume%d/%s", Config.getVolumeBasePath (), args.volume,
+				args.uuid.toString ()));
+
+			// fallback to backup
+			// NOTE: This could have problems in NFS
```
Comment (PedroChaparro) [inline app/src/main/java/worker/file/DownloadFile.java:34]: Why? Do you mean the fallback to the backup folder or the bytes streaming?

File: app/src/test/java/worker/UTFileIO.java
Comment (PedroChaparro) [inline app/src/test/java/worker/UTFileIO.java:1]: What do you think about adding another test removing the original file and streaming the one in the backup folder?

**Review by PedroChaparro — COMMENTED** [review]
It is a good idea to stream the bytes ⚡

File: app/src/main/java/worker/file/DownloadFile.java (hunk 1 above)
Comment (Woynert) [reply app/src/main/java/worker/file/DownloadFile.java:34]: I mean the function used to check whether or not a file exists. I did read some comment in a thread from about 2012 where a guy talked about it. However if it's a bug with the IO API it probably should have been fixed by now. So for now it's just a warning if anything happens.

File: app/src/test/java/worker/UTFileIO.java
Comment (Woynert) [reply app/src/test/java/worker/UTFileIO.java:1]: Good idea, added!

File: app/src/main/java/worker/file/DownloadFile.java (hunk 1 above)
Comment (PedroChaparro) [reply app/src/main/java/worker/file/DownloadFile.java:34]: Ok, thanks for warning us. Consider sharing us the thread if you find it again.

**Review by PedroChaparro — APPROVED** [review]
Good job :clap:

**Pull request merged**